config/initializers/content_security_policy.rb

```diff
@@ -19,6 +19,18 @@ def host_to_url(str)
 media_host ||= host_to_url(ENV['S3_HOSTNAME']) if ENV['S3_ENABLED'] == 'true'
 media_host ||= assets_host
 
+def sso_host
+  return if ENV['OMNIAUTH_ONLY'] != 'true' || Devise.omniauth_providers.length != 1
+
+  provider = Devise.omniauth_configs[Devise.omniauth_providers[0]]
+  @sso_host ||= begin
+    uri = provider.cas_url if ENV['CAS_ENABLED'] == 'true'
+    uri = provider.options[:idp_sso_target_url] if ENV['SAML_ENABLED'] == 'true'
+    uri = OpenIDConnect::Discovery::Provider::Config.discover!(ENV['OIDC_ISSUER']).authorization_endpoint if ENV['OIDC_ENABLED'] == 'true'
```

Comment:
While I like the idea, I'm afraid this line specifically will cause the server to fail starting up if the OIDC provider is temporarily down or unreachable.

Reply:
Yeah. You are right about that. 

Disappointingly I don't see any way around this - except arguing that an instance with this configuration will not work regardless, if the OIDC IdP isn't setup - but there may be valid reasons why the IdP is down while Mastodon is starting.

Reply:
I agree this seems to be the best that you can do under the circumstances. However, I'm still uncomfortable with that.

What about introducing another environment variable to enable this one-click behavior? This way, no surprise for existing user, but the new behavior can be opt-in and be properly documented regarding these failure modes.